USART/Core/Src/main.c

```diff
@@ -3,10 +3,10 @@
 
 int __io_putchar(int ch);	//restructured __io_putchar() function to use USART as output
 
-char data = 0;	//data received via USART
-
 int main(void)
 {
+	char data = 0;	//data received via USART
+
```

Comment:
Thinking about var names being self explanatory, It would be great to change the name "data" to something that relates better to which value it holds. Data is a very common name.